app/javascript/submission_handling/change_pickup_library.js

```diff
@@ -0,0 +1,75 @@
+// This file is basically a class
+
+import { pollFetch, reportError } from './polling'
+import each from 'lodash/map'
+
+// Attributes
+const pendingHoldsForm = document.querySelector('form#pending-holds');
+const pendingHoldsFormCheckboxes = document.querySelectorAll('form#pending-holds .checkbox');
+const spinner = `<div class="spinner-border" role="status">
+                   <span class="sr-only">Loading...</span>
+                 </div>`;
+const pickupChangeSelect = document.querySelector('[data="pickup-location"]');
+
+const getJobInfo = async function (holdId) {
```

Comment:
@cdmo do you think we can take methods like `getJobInfo` and `deleteData` out and move to a file to be used by other asyncing work as well, like polling.js idea? These seems to gonna be similar among all?
If this is sth that makes sense, we do need to do this with this PR since this is already a big PR and this is very close to be merged so we can do that in another PR.

Reply:
I think that's a great suggestion. I can do it in this PR, np.

Reply:
ok that I moved it into polling.js you think? they seem tied to me. The way I did it is that the function is just expecting you to give it a callback from whatever is calling it, and that callback will probably do some DOM manipulation (like updating the pickup library `td` cell)

Reply:
yes, polling.js works ✔️